src/RefererId/TokenGenerator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\RefererId;
+use Symfony\Component\Security\Csrf\TokenGenerator\UriSafeTokenGenerator;
+
+/**
+ * Generates a 0 characters short token for referer management in the URL.
+ *
+ * @author Yanick Witschi <https://github.com/toflar>
+ */
+
+class TokenGenerator extends UriSafeTokenGenerator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function generateToken()
+    {
+        return substr(parent::generateToken(), 0, 8);
+    }
+}
```

Comment:
Could you please, please, please go to `File -> Settings` in there go to `Editor -> General` there check "Ensure line feed at file end on Save". :)